int64 time_step